### PR TITLE
[kong] additional arguments conf for jobs

### LINKED ISSUE #386: Kong migration job args extension
Hi Kong Team,

I would like to make ENV var KONG_PASSWORD only been set or used in kong migration only so that the generated yaml is clean, since KONG_PASSWORD is only applicable during the boostrap migration job only.

Also, I would like to try to highlight about the one of the best practice, such as CKV_K8S_35 (Prefer using secrets as files over secrets as environment variables).  

The generated bootstrap job yaml is something like this:

```yaml
  containers:
  - args: [ "/bin/sh", "-c", "export KONG_PASSWORD=`cat /etc/secrets/kong/password` && kong migrations bootstrap -vv" ]
```

we can use it as default values for it.

```yaml
migrations:
     additionalArgs: ""
     additionalPreUpgradeArgs: ""
     additionalPostUpgradeArgs: "" 
```

migration template is something like this:

```yaml
  containers:
  - args: [ "/bin/sh", "-c", "{{ .Values.migrations.additionalArgs }} kong migrations bootstrap" ]
```

pre-upgrade migration template:
```yaml
  containers:
  - args: [ "/bin/sh", "-c", "{{ .Values.migrations.additionalPreUpgradeArgs }} kong migrations up" ]
```

post-upgrade migration template:
```yaml
  containers:
  - args: [ "/bin/sh", "-c", "{{ .Values.migrations.additionalPostUpgradeArgs }} kong migrations finish" ]
```

Let me know your feedback

### PR DESCRIPTION
**Pull request by ttony**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #386 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `next` branch and targets `next`, not `main`
- [x] New or modified sections of values.yaml are documented in the README.md
- [x] Title of the PR and commit headers start with chart name (e.g. `[kong]`)

### CODE REVIEW

**Comment by rainest** [comment]
Hey @ttony, pinging on this. Did you have any feedback on this re https://github.com/Kong/charts/issues/386#issuecomment-888552477 and https://github.com/Kong/charts/issues/386#issuecomment-892171572? With the proposed docker-kong change in https://github.com/Kong/docker-kong/issues/488#issuecomment-895974436, are you okay closing this for that instead?

File: charts/kong/templates/migrations-pre-upgrade.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ spec:
         {{ toYaml .Values.containerSecurityContext | nindent 10 }} 
         env:
         {{- include "kong.no_daemon_env" . | nindent 8 }}
-        args: [ "/bin/sh", "-c", "kong migrations up" ]
+        args: [ "/bin/sh", "-c", "{{ .Values.migrations.additionalPreUpgradeArgs }} kong migrations up" ]
```
Comment (ccfishk) [inline charts/kong/templates/migrations-pre-upgrade.yaml:60]:
```suggestion
        args: [ "/bin/sh", "-c", "{{ .Values.migrations.additionalPreUpgradeArgs }} kong migrations starts" ]
```

**Review by ccfishk — COMMENTED** [review]
I believe this changes are needed, would be helpful if we have tests when adding them.

**Comment by Tieske** [comment]
fyi the referenced https://github.com/Kong/docker-kong/pull/490 has been merged. But older images will not have this available.

**Comment by rainest** [comment]
Closing per https://github.com/Kong/charts/issues/386#issuecomment-911883119

**Pull request closed without merging**